packages/grid/x-data-grid/src/tests/quickFiltering.DataGrid.test.tsx

```diff
@@ -638,4 +645,48 @@ describe('<DataGrid /> - Quick Filter', () => {
       rows: [],
     });
   });
+
+  // https://github.com/mui/mui-x/issues/9677
+  it('should not fail when adding a grouping criterion', () => {
```

Comment:
Extract it to a new file `quickFiltering.DataGridPremium.test.tsx` and keep this one using the DataGrid?